lib/tasks/dev.rake

```diff
@@ -0,0 +1,89 @@
+desc "Fill the database tables with some sample data"
+task({ :sample_data => :environment}) do
+  
+  User.destroy_all
+  Loan.destroy_all
+  Adjustment.destroy_all
+
+  # Create Sample Users
+    #  email                  :string           default(""), not null
+    #  encrypted_password     :string           default(""), not null
+  email = ["alice@example.com", "bob@example.com", "carol@example.com"]
+
+  3.times do |count|
+    user = User.new
+    user.email = email.at(count)
+    user.password = "password"
+    user.save
+  end
+
+  p "Added #{User.count} Users"
+
+  # Create Sample Loans
+    #  current_balance     :float
+    #  original_amount     :float
+    #  interest_rate       :float
+    #  periods_in_year     :integer
+    #  user_id             :integer
+    #  loan_name           :string
+    #  monthly_min_payment :float
+
+    #cur_bal = [1000, 5000, 10000, 15000, 20000, 25000, 30000, 35000, 40000]
+    org_bal = [1000, 5000, 10000, 15000, 20000, 25000, 30000, 35000, 40000]
+    int_rate = [2, 3, 4, 5, 6]
+    payback_per = [2, 4, 6]
+    per_in_year = 12
+    users = User.all
+    l_name = ["Car", "Car 2", "Student Loan", "Grad School", "Credit Card"]
+
+    10.times do
+      loan = Loan.new
+      loan.current_balance = org_bal.sample
+      loan.original_amount = loan.current_balance
+      loan.interest_rate = int_rate.sample
+      loan.periods_in_year = per_in_year
+      loan.user_id = users.sample.id
+      loan.loan_name = l_name.sample
+      
+      # variables for min payments
+      rate = loan.interest_rate/100/12
+      periods = payback_per.sample * loan.periods_in_year
+
+      loan.monthly_min_payment = loan.original_amount * ( rate*(1 + rate)**periods) / ((1+rate)**periods -1)
+
+      loan.save
```

Comment:
Would it be possible to move this calculation logic to a model method and then run it automatically with an [ActiveRecord callback](https://guides.rubyonrails.org/active_record_callbacks.html), maybe a `before_save` or an `after_save`? It feels a little dangerous to have seemingly important calculations happening in the sample data file, which could go stale if the logic is updated elsewhere.

Reply:
This is what I would like to do, but I realized that my original application model/schema is missing a couple key columns to make this work correctly, specifically the original `payback period`.  I have a couple other model changes I need to make, and will get that added as part of those updates.  This is a bandaid to get a an accurate monthly payment for now but will move this calculation when I can.  Thanks for the suggestion!

Reply:
I see. For the time being, perhaps you can move most of the logic into an instance method and pass in `payback_period` as an argument, at least?

Reply:
Done.  I moved the `rate` and the `periods` to this method as well since that is the only place they are being used.  This is what this looks like now in the `loan model`

```ruby
    def monthly_min_payment_for_payback_period( payback_period )
      rate = interest_rate/100/12
      periods = payback_period * periods_in_year

      calculated_monthly_min_payment = original_amount * ( rate*(1 + rate)**periods) / ((1+rate)**periods -1)
    end
```